apps/api/src/modules/etl/events/guests/delete.ts

```diff
@@ -0,0 +1,80 @@
+import assert from 'assert';
+import Debug from 'debug';
+import { App } from 'firebase-admin/app';
+import { getAuth as getFirebaseAuth } from 'firebase-admin/auth';
+import { getFirestore as getFirebaseFirestore } from 'firebase-admin/firestore';
+
+import { EventId, UserId } from '../../../../interfaces';
+import { ApiError, makeApiError } from '../../../../../lib/errors';
+
+type Params = {
+  eventId: EventId;
+  userId: UserId;
+};
+
+type Context = {
+  firebaseClientInjection: App;
+};
+
+export const etlEventsGuestsDelete = async (
+  params: Params,
+  context: Context,
+  { debug = Debug('api:etl/events/guests/delete') as any } = {}
+) => {
+  assert(params.eventId, makeApiError(400, 'Event is required'));
+  assert(params.userId, makeApiError(400, 'User is required'));
+
+  debug(`Deleting an event: ${JSON.stringify(params, null, 4)}`);
+
+  const firebaseAuth = getFirebaseAuth(context.firebaseClientInjection);
+  const firebaseFirestore = getFirebaseFirestore(
+    context.firebaseClientInjection
+  );
+  assert(
+    firebaseAuth && firebaseFirestore,
+    makeApiError(422, 'Invalid context')
+  );
+
+  try {
+    await firebaseFirestore.runTransaction(async (transaction) => {
+      const eventDocRef = firebaseFirestore.doc(`/events/${params.eventId}`);
+      const event = (await transaction.get(eventDocRef)).data();
+
+      assert(event, makeApiError(422, 'Invalid event'));
+
+      const guestsByUserId = event.guestsByUserId as UserId[];
+      // Assert that the params.userId is a guest in the event doc
+      assert(
+        guestsByUserId.includes(params.userId),
+        makeApiError(400, `User: ${params.userId} is not a guest`)
```

Comment:
Wait my bad this should be 422 -- means the request was properly formatted but there is a semantic / logic error